### PR TITLE
PN-123 matching notification

### PR DESCRIPTION
**Pull request by itaiGershtansky**

### CODE REVIEW

File: core/client/src/main/java/org/phenotips/remote/client/internal/RemoteMatchFinder.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            }
+
+            String configuredServerId = remote.getStringValue(ApplicationConfiguration.CONFIGDOC_REMOTE_SERVER_ID);
+            if (StringUtils.isEmpty(configuredServerId)) {
+                continue;
+            }
+
+            // Include only servers that are marked for remote search
+            int searchMatches = remote.getIntValue(ApplicationConfiguration.CONFIGDOC_REMOTE_SERVER_SEARCH_MATCHES);
+            if (searchMatches != 1) {
```
Comment (buske) [inline core/client/src/main/java/org/phenotips/remote/client/internal/RemoteMatchFinder.java]: Shouldn't this just be `!searchMatches`?

File: core/client/src/main/java/org/phenotips/remote/client/script/RemoteMatchingScriptService.java [hunk 2]
```diff
@@ -17,23 +17,24 @@
...
+
 import javax.inject.Inject;
 import javax.inject.Named;
 import javax.inject.Singleton;
 
-import org.slf4j.Logger;
-
 import org.json.JSONArray;
 import org.json.JSONObject;
+import org.slf4j.Logger;
```
Comment (buske) [inline core/client/src/main/java/org/phenotips/remote/client/script/RemoteMatchingScriptService.java]: More import lines moving around

File: core/common/src/main/java/org/phenotips/remote/common/internal/RemotePatientSimilarityView.java [hunk 3]
```diff
@@ -17,14 +17,14 @@
...
-import org.json.JSONObject;
-
 import org.phenotips.data.Patient;
 import org.phenotips.data.similarity.AccessType;
 import org.phenotips.data.similarity.internal.DefaultPatientSimilarityView;
 import org.phenotips.remote.api.ContactInfo;
 import org.phenotips.remote.api.MatchingPatient;
 
+import org.json.JSONObject;
+
```
Comment (buske) [inline core/common/src/main/java/org/phenotips/remote/common/internal/RemotePatientSimilarityView.java]: Moving imports shouldn't be part of PR

File: core/client/src/main/java/org/phenotips/remote/client/RemoteMatchingService.java [hunk 4]
```diff
@@ -17,15 +17,14 @@
...
-
-import org.phenotips.data.similarity.PatientSimilarityView;
-import org.phenotips.remote.api.ApiViolationException;
 import org.phenotips.remote.api.OutgoingMatchRequest;
+import org.phenotips.remote.common.internal.RemotePatientSimilarityView;
 
 import org.xwiki.component.annotation.Role;
 import org.xwiki.stability.Unstable;
 
+import java.util.List;
```
Comment (sdumitriu) [inline core/client/src/main/java/org/phenotips/remote/client/RemoteMatchingService.java]: The new order is correct. Not sure why it wasn't here before...

File: core/client/src/main/java/org/phenotips/remote/client/internal/RemoteMatchFinder.java (hunk 1 above)
Comment (sdumitriu) [inline core/client/src/main/java/org/phenotips/remote/client/internal/RemoteMatchFinder.java]: In Java, you can't use `!integer`, types are stricter than in JavaScript.

Comment (buske) [inline core/client/src/main/java/org/phenotips/remote/client/internal/RemoteMatchFinder.java]: Ack, sorry!

**Pull request merged**